### ISSUE TITLE
Bug(parser): Exported classes exclude decorators in their location properties

### ISSUE THREAD

**Issue by DMartens** [typescript-eslint/typescript-eslint#8769]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.2&showAST=es&fileType=.tsx&code=AIOw9gLgkiDGA2BXAJgU2TASgQxAc1QChUAPABzACcIACBbAZwZoBEaBvGgX0MOAEs4SNBhA58RekxoBhDt15A&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
@decorator
export class Class {}
```

### ESLint Config

```javascript
Not relevant
```

### tsconfig

```jsonc
Not relevant
```

### Expected Result

I expect that the node `range` and `loc` of exported classes to include the decorators.
This is the case if there is no `export` keyword (e.g. `@decorator class Class {}`).

### Actual Result

The node `range` and `loc` starts at the `export` keyword (in case of the example on line 2).

### Additional Info

`@babel/eslint-parser` currently also does not correctly reflect this and I see no discussion about whether the decorators should be included in the range in the estree issue tracker.
I expect this bug exists as decorators were previously not allowed before the `export` keyword.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/parser`            | `7.3.1` |

**Comment by bradzacher** [comment]
This is working as intended right now.

The AST is a little weird and there's not really a good representation for this.
TS gets away with it because it includes the `export` as part of the class's node - so it can include the decorator as well without having anything weird.

OTOH you can't do the same thing with the estree AST - the decorator exists before the export keyword - and the export keyword forms a parent node to the class.

This is something that we left as is with the initial implementation and has worked well enough. But it is a messy thing that is not documented or specced out.

Filed this to try and improve the spec clarity. https://github.com/estree/estree/issues/315
